pkg/commands/drift/statefiles/statefiles.go

```diff
@@ -163,33 +154,19 @@ func (c *DriftStatefilesCommand) Process(ctx context.Context) error {
 			errs = append(errs, fmt.Errorf("unsupported backend type for terraform config at %s - only gcs backends are supported", f))
 			continue
 		}
-		gcsBuckets = append(gcsBuckets, *config.GCSBucket)
+		gcsBuckets[*config.GCSBucket] = struct{}{}
 		expectedURIs = append(expectedURIs, fmt.Sprintf("gs://%s/%s/default.tfstate", *config.GCSBucket, *config.Prefix))
 	}
 	if len(errs) > 0 {
 		return fmt.Errorf("failed to determine statefile gcs URIs: %w", errors.Join(errs...))
 	}
 
-	gcsBucket := c.flagGCSBucket
-	for _, bucket := range gcsBuckets {
-		if gcsBucket == "" {
-			gcsBucket = bucket
-		}
-		if bucket != gcsBucket {
-			return fmt.Errorf("found multiple definitions for gcs buckets - expected a single gcs bucket; from configs: %v; expected %s", gcsBuckets, gcsBucket)
-		}
-	}
-
-	if gcsBucket == "" {
-		return fmt.Errorf("unable to determine gcs bucket - please provide the gcsBucket flag or point at a terraform config with gcs backends")
-	}
-
-	logger.DebugContext(ctx, "finding statefiles in gcs bucket",
-		"gcs_bucket", gcsBucket)
+	logger.DebugContext(ctx, "finding statefiles in gcs buckets",
```

Comment:
nit: `gcsBuckets` is a map and the output will look like "map[bar:{} foo:{}]". Instead, right before this line, we could turn it into a slice and just use the slice below.

Reply:
Done